### PR TITLE
Improve sanitization of user submitted data in $_SERVER

### PR DESCRIPTION
**Pull request by Tabrisrp**

### CODE REVIEW

File: min/config.php [hunk 1]
```diff
@@ -8,10 +8,11 @@
 
 // load WP Rocket config file
 $host = $_SERVER['HTTP_HOST'];
-$host = str_replace( array( '..', chr(0) ), '', $host );
-$wp_rocket_config_file = dirname( dirname( dirname( dirname( __FILE__ ) ) ) ) . '/wp-rocket-config/' . $host . '.php';
+$host = urlencode( $host );
+$wp_rocket_config_path = dirname( dirname( dirname( dirname( __FILE__ ) ) ) ) . '/wp-rocket-config/';
+$wp_rocket_config_file = realpath( dirname( dirname( dirname( dirname( __FILE__ ) ) ) ) . '/wp-rocket-config/' . $host . '.php' );
```
Comment (Screenfeed) [inline min/config.php]: Simpler:
```$wp_rocket_config_file = realpath( $wp_rocket_config_path . $host . '.php' );```

File: inc/front/process.php [hunk 2]
```diff
@@ -41,25 +41,25 @@
 $rocket_config_path = WP_CONTENT_DIR . '/wp-rocket-config/';
 $host = ( isset( $_SERVER['HTTP_HOST'] ) ) ? $_SERVER['HTTP_HOST'] : time();
 $host = trim( strtolower( $host ), '.' );
-$host = str_replace( array( '..', chr( 0 ) ), '', $host );
+$host = urlencode( $host );
 
 $continue = false;
-if ( file_exists( $rocket_config_path . $host . '.php' ) ) {
+if ( file_exists( realpath( $rocket_config_path . $host . '.php' ) ) && false !== stripos( realpath( $rocket_config_path . $host . '.php' ), $rocket_config_path ) ) {
```
Comment (Screenfeed) [inline inc/front/process.php]: I would replace all `false !== stripos(` by `0 === stripos(`. Don't you agree?

Comment (Tabrisrp) [inline inc/front/process.php]: yes agreed we expect the needle at the start so we can do that.

File: min/config.php (hunk 1 above)
Comment (Tabrisrp) [inline min/config.php]: 👍

File: min/config.php [hunk 3]
```diff
@@ -8,10 +8,11 @@
...
 // load WP Rocket config file
 $host = $_SERVER['HTTP_HOST'];
-$host = str_replace( array( '..', chr(0) ), '', $host );
-$wp_rocket_config_file = dirname( dirname( dirname( dirname( __FILE__ ) ) ) ) . '/wp-rocket-config/' . $host . '.php';
+$host = urlencode( $host );
+$wp_rocket_config_path = dirname( dirname( dirname( dirname( __FILE__ ) ) ) ) . '/wp-rocket-config/';
+$wp_rocket_config_file = realpath( $wp_rocket_config_path . $host . '.php' );
 
-if ( file_exists( $wp_rocket_config_file ) && ! defined( 'ABSPATH' ) ) {
+if ( file_exists( $wp_rocket_config_file ) && false !== stripos( $wp_rocket_config_file, $wp_rocket_config_path ) && ! defined( 'ABSPATH' ) ) {
```
Comment (Screenfeed) [inline min/config.php]: You forgot this one :D

**Pull request merged**